### PR TITLE
Add Nginx h2c rule and various ASVS metadata

### PR DESCRIPTION
**Pull request by hazanasec**
Hi guys,

Added a rule to detect possible h2c smuggling in nginx, and added a bunch more ASVS metadata to existing rules.

Thanks,
Hazana

### CODE REVIEW

File: generic/nginx/security/possible-h2c-smuggling.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+rules:
+- id: possible-h2c-smuggling
+  patterns:
+    - pattern-either: 
+      - pattern: |
+          proxy_http_version 1.1 ...; ... proxy_set_header Upgrade ...; ... proxy_set_header Connection ...;
```
Comment (mschwager) [inline generic/nginx/security/possible-h2c-smuggling.yaml]: I think these will be easier to read if we place them on multiple lines, e.g.

```
proxy_http_version 1.1 ...;
...
proxy_set_header Upgrade ...;
...
```

**Review by mschwager — COMMENTED** [review]
Nice! Thanks for the new rule and ASVS information :+1:

I left one comment, and after that this LGTM

File: java/lang/security/insecure-jms-deserialization.yaml [hunk 2]
```diff
@@ -6,8 +6,13 @@ rules:
...
     cwe: 'CWE-502: Deserialization of Untrusted Data'
     owasp: 'A8: Insecure Deserialization'
+    asvs:
+      section: "V5 Validation, Sanitization and Encoding"
+      control_id: "5.5.3 Insecue Deserialization"
+      control_url: "https://github.com/OWASP/ASVS/blob/master/4.0/en/0x13-V5-Validation-Sanitization-Encoding.md#v55-deserialization-prevention"
+      version: "4"    
     references:
-    - https://www.blackhat.com/docs/us-16/materials/us-16-Kaiser-Pwning-Your-Java-Messaging-With-Deserialization-Vulnerabilities-wp.pdf
+    - https://www.blackhat.com/docs/us-16/materials/us-16-Kaiser-Pwning-Your-Java-Messaging-With-Deserialization-Vulnerabilities-wp.pdf
```
Comment (mschwager) [inline java/lang/security/insecure-jms-deserialization.yaml]:
```suggestion
    - https://www.blackhat.com/docs/us-16/materials/us-16-Kaiser-Pwning-Your-Java-Messaging-With-Deserialization-Vulnerabilities-wp.pdf
```

**Review by mschwager — APPROVED** [review]
:tada:

**Comment by mschwager** [comment]
Looks like `pre-commit` is complaining about some trailing whitespace issues. We should be good to go after those are fixed :+1:

**Comment by hazanasec** [comment]
@mschwager I don't think it's displaying for me? More than happy for you to edit it

**Comment by mschwager** [comment]
I can't easily push changes to this PR because the remote origin is your fork. So instead I created https://github.com/returntocorp/semgrep-rules/pull/1264 :+1:

**Comment by hazanasec** [comment]
awesome, thanks

**Pull request closed without merging**